Ginger/GingerCoreTest/Misc/DataSourceTest.cs

```diff
@@ -0,0 +1,276 @@
+            DataSourceTable dataSource = null;
+            List<string> mColumnNames = null;
+            foreach (DataSourceTable dataSourceTable in dataSourceTableList)
+            {
+                if (dataSourceTable.Name == "MyCustomizedDataTable")
+                {
+                    dataSource = dataSourceTable;
+                }
+            }
+            DataTable dataTable = accessDataSource.GetQueryOutput(dataSource.Name);
+            dataSource.DataTable = dataTable;
+            accessDataSource.AddRow(mColumnNames, dataSource);
+            dataTable = accessDataSource.GetQueryOutput(dataSource.Name);
+
+            //Act
+            accessDataSource.SaveTable(dataTable);
+            var a = accessDataSource.GetQueryOutput("select * from MyCustomizedDataTable");
+
+            //Assert
+            Assert.AreEqual("1", a, "RowCount");
+        }
+
+
+        [TestMethod]
+        public void CopyTable()
+        {
+            // Assert
+            string TableName = "MyCustomizedDataTable";
+            string NewTablename = TableName + "_Copy";
+
+            //Act
+            accessDataSource.CopyTable(TableName);
+
+            //Assert
+            Assert.AreEqual(accessDataSource.IsTableExist(NewTablename), true);
+
+            //Act : delete table
+            accessDataSource.DeleteTable(NewTablename);
+
+            //Assert
+            Assert.AreEqual(accessDataSource.IsTableExist(NewTablename), false);
+        }
+
+        [TestMethod]
+        public void GetQueryOutput()
+        {
+            //Arrange
+            string Query = "select * from MyCustomizedDataTable";
+
+            //Act
+            DataTable res = accessDataSource.GetQueryOutput(Query);
+
+            //Assert
+            Assert.AreEqual(1, res.Rows.Count);
+        }
+
+        [TestMethod]
+        public void ExportToExcel()
+        {
+            //Arrange
+            string Query = "select * from MyCustomizedDataTable";
+
+            //Act
+            DataTable dt = accessDataSource.GetQueryOutput(Query);
+            bool res = GingerCoreNET.GeneralLib.General.ExportToExcel(dt, excelFilePath, "accessDataSourceExportedData");
+
+            //// TODO:  Fetch and validate data from exported excel using excel action
+
+            //Assert
+            Assert.AreEqual(true, res);
+        }
+
+        [TestMethod]
+        public void CommitDb()
+        {
+            // Arrange
+            ObservableList<DataSourceTable> dataSourceTableList = accessDataSource.GetTablesList();
+            DataSourceTable dataSource = null;
+            List<string> mColumnNames = null;
+            foreach (DataSourceTable dataSourceTable in dataSourceTableList)
+            {
+                if (dataSourceTable.Name == "MyCustomizedDataTable")
+                {
+                    dataSource = dataSourceTable;
+                }
+            }
+            DataTable dataTable = accessDataSource.GetQueryOutput(dataSource.Name);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['dataSource' is null on at least one execution path.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=4855725)